Update dep

EagleEye25 opened the pull request.

Commit:
updated initial dependancies

Commit:
update according to angular steps

Commit:
changed everything to required by versions

Commit:
more updates to deps

Commit:
fix on change for preview listing

Commit:
updated all tab change events

Commit:
fix paginator bug

Commit:
fix selects

Commit:
fix mat-select-search

Commit:
removed all console.log statements

Commit:
added last pipe

Commit:
fix fxLayoutWrap

Commit:
Merge branch 'market' into update-dep

Commit:
fix linting issues

Commit:
remove package-lock.json

vikas-cis-zz commented on src/app/wallet/wallet/receive/receive.component.html:
```diff
@@ -132,16 +132,6 @@
-      <app-paginator
-        *ngIf="type == 'public' && getSinglePage().length > 0"
         #paginator
         [length]="getTotalCountForPagination()"
```
I believe `getSinglePage().length > 0` this should be their as it only show the pagination when data is available

EagleEye25 replied:
changes has been completed

zaSmilingIdiot replied:
Ah, yes, thanks for spotting that @vikas-cis  :)

@EagleEye25 please include the condition for app-paginator for
`*ngIf="getSinglePage().length > 0"`

vikas-cis-zz reviewed (changes requested):
Is it needed to remove package-lock ?

Commit:
minor fix for paginator

zaSmilingIdiot committed:
Update package.json

Updates for building prod environments using angular 7

zaSmilingIdiot reviewed (changes requested):
When building the application ( using `yarn package:linux` for example), the build fails with the following errors, which can and need to all be corrected:

```
ERROR in src/app/modals/shared/daemon-connection/daemon-connection.component.html(1,22): : Property 'txt' does not exist on type 'DaemonConnectionComponent'. Did you mean 'text'?
src/app/core-ui/main/cart/cart.component.html(6,9): : Property 'listCache' is private and only accessible within class 'CartComponent'.
src/app/core-ui/main/cart/cart.component.html(6,9): : Property 'listCache' is private and only accessible within class 'CartComponent'.
src/app/core-ui/main/cart/cart.component.html(9,9): : Property 'listCache' is private and only accessible within class 'CartComponent'.
src/app/core-ui/main/cart/cart.component.html(9,9): : Property 'listCache' is private and only accessible within class 'CartComponent'.
src/app/core-ui/paginator/paginator.component.html(2,46): : Property 'itemsPerPageLabel' does not exist on type 'Object'.
src/app/core-ui/paginator/paginator.component.html(23,3): : Property 'firstPageLable' does not exist on type 'Object'.
src/app/core-ui/paginator/paginator.component.html(30,3): : Property 'previousPageLabel' does not exist on type 'Object'.
src/app/core-ui/paginator/paginator.component.html(37,3): : Property 'nextPageLabel' does not exist on type 'Object'.
src/app/core-ui/paginator/paginator.component.html(44,3): : Property 'lastPageLable' does not exist on type 'Object'.
src/app/wallet/wallet/send/send.component.html(42,70): : Argument of type '"part"' is not assignable to parameter of type 'TxType'.
src/app/wallet/wallet/send/send.component.html(46,55): : Argument of type '"blind"' is not assignable to parameter of type 'TxType'.
src/app/wallet/wallet/send/send.component.html(47,25): : Argument of type '"blind"' is not assignable to parameter of type 'TxType'.
src/app/wallet/wallet/shared/qr-code-modal/qr-code-modal.component.html(18,41): : Expected 1 arguments, but got 0.
src/app/market/listings/listing-item/listing-item.component.html(22,10): : Property 'listingCacheService' is private and only accessible within class 'ListingItemComponent'.
src/app/market/listings/listing-item/listing-item.component.html(22,10): : Property 'listingCacheService' is private and only accessible within class 'ListingItemComponent'.
src/app/market/listings/listing-item/listing-item.component.html(28,10): : Property 'listingCacheService' is private and only accessible within class 'ListingItemComponent'.
src/app/market/listings/listing-item/listing-item.component.html(28,10): : Property 'listingCacheService' is private and only accessible within class 'ListingItemComponent'.
src/app/market/listings/preview-listing/preview-listing.component.html(80,29): : Property 'countryListService' is private and only accessible within class 'PreviewListingComponent'.
src/app/market/listings/preview-listing/preview-listing.component.html(80,29): : Property 'countryListService' is private and only accessible within class 'PreviewListingComponent'.
src/app/market/listings/listings.component.html(17,9): : Property 'countryList' is private and only accessible within class 'ListingsComponent'.
src/app/market/listings/listings.component.html(17,9): : Property 'countryList' is private and only accessible within class 'ListingsComponent'.
src/app/market/listings/listings.component.html(38,9): : Property 'countryList' is private and only accessible within class 'ListingsComponent'.
src/app/market/listings/listings.component.html(38,9): : Property 'countryList' is private and only accessible within class 'ListingsComponent'.
src/app/market/sell/add-item/add-item.component.html(93,9): : Property 'countryList' is private and only accessible within class 'AddItemComponent'.
src/app/market/buy/checkout-process/checkout-process.component.html(40,17): : Property 'listCache' is private and only accessible within class 'CheckoutProcessComponent'.
src/app/market/buy/checkout-process/checkout-process.component.html(40,17): : Property 'listCache' is private and only accessible within class 'CheckoutProcessComponent'.
src/app/market/buy/checkout-process/checkout-process.component.html(108,59): : Property 'listCache' is private and only accessible within class 'CheckoutProcessComponent'.
src/app/market/buy/checkout-process/checkout-process.component.html(108,59): : Property 'listCache' is private and only accessible within class 'CheckoutProcessComponent'.
src/app/market/buy/checkout-process/checkout-process.component.html(118,19): : Property 'listCache' is private and only accessible within class 'CheckoutProcessComponent'.
src/app/market/buy/checkout-process/checkout-process.component.html(118,19): : Property 'listCache' is private and only accessible within class 'CheckoutProcessComponent'.
src/app/market/buy/checkout-process/checkout-process.component.html(121,19): : Property 'listCache' is private and only accessible within class 'CheckoutProcessComponent'.
src/app/market/buy/checkout-process/checkout-process.component.html(121,19): : Property 'listCache' is private and only accessible within class 'CheckoutProcessComponent'.
src/app/market/buy/checkout-process/checkout-process.component.html(260,13): : Property 'confirmation' does not exist on type 'CheckoutProcessComponent'.
src/app/market/shared/orders/order-item/order-item.component.html(11,17): : Property 'hash_bg' does not exist on type 'Bid'.
src/app/market/shared/orders/order-item/order-item.component.html(11,75): : Property 'hash' does not exist on type 'Bid'.
src/app/market/shared/orders/order-item/order-item.component.html(85,57): : Property 'show_escrow_txdetails' does not exist on type 'Bid'.
src/app/market/shared/report/report.component.html(1,11): : This condition will always return 'false' since the types 'boolean' and '"listing"' have no overlap.
src/app/market/shared/report/report.component.html(9,9): : This condition will always return 'false' since the types 'boolean' and '"preview"' have no overlap.
src/app/wallet/proposals/add-proposal/add-proposal.component.html(70,19): : Property 'length' does not exist on type 'AbstractControl'.
src/app/wallet/proposals/add-proposal/add-proposal.component.html(80,11): : Property 'controls' does not exist on type 'AbstractControl'.
src/app/wallet/overview/widgets/coldstake/coldstake.component.html(6,13): : Property '_coldstake' is private and only accessible within class 'ColdstakeComponent'.
src/app/wallet/overview/widgets/coldstake/coldstake.component.html(11,13): : Property '_coldstake' is private and only accessible within class 'ColdstakeComponent'.
src/app/wallet/overview/widgets/coldstake/coldstake.component.html(11,13): : Property '_coldstake' is private and only accessible within class 'ColdstakeComponent'.
src/app/wallet/overview/widgets/coldstake/coldstake.component.html(19,13): : Property '_coldstake' is private and only accessible within class 'ColdstakeComponent'.
src/app/wallet/overview/widgets/coldstake/coldstake.component.html(19,13): : Property '_coldstake' is private and only accessible within class 'ColdstakeComponent'.
src/app/wallet/overview/widgets/coldstake/coldstake.component.html(27,26): : Property '_coldstake' is private and only accessible within class 'ColdstakeComponent'.
src/app/wallet/overview/widgets/coldstake/coldstake.component.html(27,26): : Property '_coldstake' is private and only accessible within class 'ColdstakeComponent'.
src/app/wallet/overview/widgets/coldstake/coldstake.component.html(43,13): : Property '_coldstake' is private and only accessible within class 'ColdstakeComponent'.
src/app/wallet/overview/widgets/coldstake/coldstake.component.html(43,13): : Property '_coldstake' is private and only accessible within class 'ColdstakeComponent'.
src/app/wallet/overview/widgets/coldstake/coldstake.component.html(67,26): : Property '_coldstake' is private and only accessible within class 'ColdstakeComponent'.
src/app/wallet/overview/widgets/coldstake/coldstake.component.html(73,13): : Property '_coldstake' is private and only accessible within class 'ColdstakeComponent'.
src/app/wallet/overview/widgets/coldstake/coldstake.component.html(73,13): : Property '_coldstake' is private and only accessible within class 'ColdstakeComponent'.

```

EagleEye25 commented:
I'll fix these all tomorrow morning

Commit:
fix build issues

Commit:
removed console.log

Commit:
fix small issue

Commit:
error fix

EagleEye25 commented:
just fixing a last error that showed up

Commit:
fix issue on paginator

Commit:
fix show hash

EagleEye25 commented:
@AllienWorks, would you please be able to do some styling on the display of the hash for us :)

Commit:
updates as requested

Commit:
update to bid modal

AllienWorks commented:
@EagleEye25 can we truncate the hash to first (or last) ~3-5 characters? I doubt we need the full hash. truncuated hash (kinda order ID) would be more useful and usable for users.

This PR will need _a lot_ more than styling the hashes :D the styling and layouts are messed everywhere.. should I take care of it @HenkSwardt? (this will take some considerable time).

EagleEye25 commented:
@AllienWorks I will truncate the hash to 5 chars for you :), and yeah, I tired to fix some stuff, but I'm not a css pro like you ;)

AllienWorks commented:
@EagleEye25 no need to fix CSS by you, leave that to me completely ;)
Otherwise there won't be anything left for me to do, hehe.

Btw truncate it to 4 chars max. That's IMO enough.

EagleEye25 commented:
@AllienWorks haha I got you :p, awesome will do that tomorrow morning for you first thing

Commit:
truncated hash

EagleEye25 commented:
@AllienWorks the hash has been truncated :D

HenkSwardt commented:
@AllienWorks , yes please, 

Lets get this added. 

Thank you

Commit:
update to hash truncate

Commit:
styling: TX table

Commit:
styling: tabs, expansion panels

Commit:
styling: main sidenav, status bar

Commit:
styling: cards, buttons, lists..

zaSmilingIdiot commented on src/app/wallet/wallet/send/send.component.ts:
```diff
@@ -36,6 +37,9 @@ export class SendComponent implements OnInit {
   public send: TransactionBuilder;
+  part: TxType.PUBLIC;
+  blind: TxType.BLIND;
+  anon: TxType.ANON;
```
The declaration of these 3 is unnecessary: it is possible to declare this instead:
`public TxType: typeof TxType = TxType;`
This makes the enum type available in the template as well (allows for using `TxType.BLIND` for example in the template).

zaSmilingIdiot commented on src/app/wallet/wallet/send/send.component.html:
```diff
@@ -39,18 +39,18 @@
           <mat-radio-button *ngIf="testnet" class="balance" value="anon" color="primary" (change)="setInputOutput('anon', 'output')" fxFlex >
             <div class="name">Anon</div>
-            <div class="desc">Available balance:<span class="amount">{{ getBalance('anon') }}</span></div>
+            <div class="desc">Available balance:<span class="amount">{{ getBalance(anon) }}</span></div>
```
When making a change like this, make sure that the rest of the function calls are updated in a similar way. 
So for example: 
The code is using a variable/const here, but on the following line (and there are more like it in this same template file), there are string arguments: 
https://github.com/particl/particl-desktop/pull/1335/files#diff-93865e09e6daa9faca449ba9b6f543e9R135
Its best to make sure that we have consistency.

zaSmilingIdiot replied:
Also see reference to the comment on using the enum in the template

zaSmilingIdiot commented on src/app/wallet/proposals/add-proposal/add-proposal.component.html:
```diff
@@ -67,7 +67,7 @@ <h1>New Community Proposal</h1>
         <div class="title">
           Voting options
-          <button [disabled]="proposalFormGroup.controls.options.length >= 5" (click)="addNewInputField()" mat-button color="basic" class="small">
+          <button [disabled]="checkProposalItems()" (click)="addNewInputField()" mat-button color="basic" class="small">
```
There doesn't seem to be a point in breaking this out into a separate method. The basis for the method is checking:
`proposalFormGroup.controls.options.value.length >= 5`
whereas the previous check here was checking
`proposalFormGroup.controls.options.length >= 5`

The difference seems to essentially be checking for `options.value.length` vs checking for `options.length`. Moving it into a separate method call and numControls variable (apart from performance reasons) is that the numControls variable is not initialized correctly. Which means using it before any call to checkProposalItems() is going to result in an error if its expected that it contains the proposal items. In other words, `checkProposalItems()` is not just doing evaluation as its name suggests, but initialization as well. In other words, the method has additional behaviour than simply performing a check. 

It seems the process was changed here because of errors during the build step (using AOT instead of JIT): this can be fixed in the original code by changing
`proposalFormGroup.controls.options.length >= 5`
to
`proposalFormGroup.controls.options['length'] >= 5`

and changing (below)
`let option of proposalFormGroup.controls.options.controls; let i = index`
to 
`let option of proposalFormGroup.controls.options['controls']; let i = index`

zaSmilingIdiot commented on src/app/market/buy/checkout-process/checkout-process.component.ts:
```diff
@@ -56,6 +57,7 @@ export class CheckoutProcessComponent implements OnInit, OnDestroy {
   // stepper form data
   public cartFormGroup: FormGroup;
   public shippingFormGroup: FormGroup;
+  public confirmation: FormGroup;
```
Is this used anywhere?

zaSmilingIdiot replied:
Yep, it is... my bad. Ignore this comment.

zaSmilingIdiot commented on src/app/core/market/api/bid/bid.model.ts:
```diff
@@ -65,6 +65,13 @@ export class Bid extends Product {
+    if (this.order.ListingItem && this.order.ListingItem.hash) {
+      return this.order.ListingItem.hash.substring(0, 4);
+    }
+    return undefined;
```
Rather return the empty string here since the return type for the getter is 'string'.

zaSmilingIdiot commented on src/app/core-ui/paginator/intl-model.ts:
```diff
@@ -0,0 +1,7 @@
+export enum Intl {
```
This is an enum, which is different to a model. Extracting the string is a good idea, but for now, since this is only done on the paginator component, it might be better to move this enum directly into the component.

zaSmilingIdiot commented on src/app/wallet/wallet/shared/qr-code-modal/qr-code-modal.component.html:
```diff
@@ -15,7 +15,7 @@
   <div class="left">
 
-    <button mat-button class="explorer" (click)="rpcLabelUpdate()">
+    <button mat-button class="explorer" (click)="rpcLabelUpdate($event)">
```
`rpcLabelUpdate()` expects a string argument; passing in the event object is probably not a good idea - if there is a requirement to pass a parameter in, then it should be the empty string.

Commit:
fix send component.html as requested

Commit:
further fix on send.html

Commit:
fix to send.html

Commit:
fix proposals

Commit:
further fix to send.component

Commit:
fix bid send

zaSmilingIdiot committed:
Update add-proposal.component.ts

Resets unnecessary changes

Commit:
fix intl

Commit:
Merge remote-tracking branch 'origin/update-dep' into update-dep

Commit:
fix qr code modal

Commit:
Merge branch 'market' into update-dep

Commit:
Merge branch 'market' into update-dep

Commit:
Fixes linting issues and merge conflict resolution problems.

Commit:
Attempting fix at error occurring during test runs relating to a missing property 'id' on an undefined object.

Commit:
Merge branch 'market' into update-dep

Commit:
Merge branch 'market-beta' into update-dep

Commit:
fix tests

Commit:
Delete yarn.lock

Commit:
fix tests?

Commit:
Merge branch 'market-beta' into update-dep

Commit:
Merge branch 'market' into update-dep

zaSmilingIdiot committed:
Fixes merge conflicts
Merge branch 'market' of https://github.com/particl/particl-desktop into update-dep

zaSmilingIdiot committed:
Re-adding updated yarn.lock file.

zaSmilingIdiot committed:
Fixes dependency changes.

zaSmilingIdiot committed:
Fix merge conflicts while merging in branch 'market'
Merge branch 'market' of https://github.com/particl/particl-desktop into update-dep

zaSmilingIdiot committed:
Merge branch 'market' of https://github.com/particl/particl-desktop into update-dep

zaSmilingIdiot committed:
Attempts update to travis build environment (distro update to include updated packages).

zaSmilingIdiot committed:
Add xvfb to travis install/setup.

zaSmilingIdiot committed:
Attempted fix at starting xvfb on xenial in travis

zaSmilingIdiot committed:
Removes xvfb from travis, using chrome headless instead.

zaSmilingIdiot committed:
Fix headless mode specification for travis tests.

zaSmilingIdiot committed:
Reverting back to using xvfb in travis.
Following travis updates in https://docs.travis-ci.com/user/build-environment-updates/2019-01-14/

zaSmilingIdiot committed:
Removes specification of chromium-browser in travis config (no longer installed by default in xenial container).
If this executes successfully, the preferred option is to get chromium installed and use that.
In the meanwhile, this change is to attempt to actually get the tests excecuted.

zaSmilingIdiot committed:
(Attempt at) reverting back to chromium instead of chrome for travis testing.

zaSmilingIdiot committed:
Corrects dependencies installation in xenial dist from travis.

zaSmilingIdiot committed:
Sets angular testing to production mode.

Commit:
Listings > Report: fixed toggle button styling

zaSmilingIdiot committed:
Merge branch 'market' of https://github.com/particl/particl-desktop into update-dep

zaSmilingIdiot committed:
Fixes rxjs issues between merged old and new code.

AllienWorks committed:
Listing > Detail: styling fixes + lightbox

AllienWorks committed:
Order: hash tweaks & layout

AllienWorks committed:
Order: added detailed hash order info

AllienWorks commented:
Since the order hashes were a part of this PR, I'll write it here. Stop me right away if it's BS:

- truncated the default order hash to only 3 characters
- added a new `order.hashDetail`, which is the same as above, but with 6 characters - that is used to grab custom color for the hash
  - important question: is the hash "hex-compatible"? if not, it won't work..
  - I've tested this with 4 orders (max. I can do with what I see on MP currently)

![Screenshot from 2019-03-12 17-52-09](https://user-images.githubusercontent.com/1965795/54219717-51662800-44f0-11e9-964e-d41a1c094a93.png)

**other than that..**
- we now have fancy photo [lightbox for fullscreen preview](https://i.imgur.com/yxsg2EN.mp4)!
- more styling & layout tweaks obviously included :)

zaSmilingIdiot committed:
Fixes testing environment config (missing environment avrs from bad merge conflict resolution).

AllienWorks commented:
I'm not sure if it's related to updated dependencies, but working on this PR, each time I make a change, the daemon won't start after the live reload. So it's needed to close the wallet and rerun with `yarn run start..`.

AllienWorks committed:
Sell > Listings/Order & confirmation modals: tweaks | linting

AllienWorks committed:
Material: styling tweaks

AllienWorks commented:
@zaSmilingIdiot I'm done. Hopefully I've went through everything. Other than the annoying issue described in previous comment, works nicely.

zaSmilingIdiot commented:
Awesome, thanks. 

The annoying issue is a side effect of a change made in market-beta branch last week. I'll fix it up shortly.

zaSmilingIdiot committed:
Merge branch 'market' of https://github.com/particl/particl-desktop into update-dep
